raw.go

```diff
@@ -31,3 +31,32 @@ type RawSpan struct {
 	// The span's associated baggage.
 	Baggage map[string]string // initialized on first use
 }
+
+// SpanDetail summarizes the identifying fields of a Span, for
+// diagnostic and implementation-specific purposes.
+type SpanDetail struct {
```

Comment:
well, you seem to have a conflict of goals, you want to keep the fields private, yet you need to access them for some reason. Interface provides a projection, controlled access. But worth asking why there's a conflict in the first place :-)